File: 1/1.rb
class DayOne
  def part_one(filename)
    File.read(filename).split("\n\n").map do |cals|
      cals.split("\n").map(&:to_i).sum
    end.max
  end

  def part_two(filename)
    File.read(filename).split("\n\n").map do |cals|
      cals.split("\n").map(&:to_i).sum
    end.sort.reverse.slice(0..2).sum
  end
end

File: 1/1.ex
defmodule DayOne do
  def part_one do
    total_cals()
      |> Enum.max
  end

  def part_two do
    total_cals()
      |> Enum.sort
      |> Enum.reverse
      |> Enum.take(3)
      |> Enum.sum
  end

  def total_cals do
    {:ok, inputs} = File.read("./input.txt")
    inputs 
      |> String.split("\n\n")
      |> Enum.map(&String.split/1)
      |> Enum.map(fn(cals) -> 
        Enum.map(cals, fn(cal) ->
          {int,_} = Integer.parse(cal)
          int
        end)
      end)
      |> Enum.map(&Enum.sum/1)
  end
end

IO.puts DayOne.part_one
IO.inspect DayOne.part_two